Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,264 @@
+        /// Adds all prefabs and scene assets found on the assets folder to the list of objects to be migrated
+        /// </summary>
+        public void TryAddProjectForMigration()
+        {
+            AddAllAssetsOfTypeForMigration(new Type[] { typeof(GameObject), typeof(SceneAsset) });
+        }
+
+        /// <summary>
+        /// Removes object from the list of objects to migrated
+        /// </summary>
+        /// <param name="selectedObject">Object to be removed</param>
+        public void RemoveObjectForMigration(Object selectedObject)
+        {
+            migrationObjects.Remove(selectedObject);
+        }
+
+        /// <summary>
+        /// Clears list of objects to be migrated
+        /// </summary>
+        public void ClearMigrationList()
+        {
+            migrationObjects.Clear();
+        }
+
+        /// <summary>
+        /// Migrates all objects from list of objects to be migrated using the selected IMigrationHandler implementation. 
+        /// </summary>
+        /// <param name="type">A type that implements IMigrationhandler</param>
+        /// <returns></returns>
+        public bool MigrateSelection(Type type, bool askToSaveCurrentScene)
+        {
+            if (!SetMigrationHandlerInstance(type))
+            {
+                return false;
+            }
+
+            if (askToSaveCurrentScene && !EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo())
+            {
+                    return false;
+            }
+            var previousScenePath = EditorSceneManager.GetActiveScene().path;
+
+            for (int i = 0; i <  migrationObjects.Count; i++)
+            {
+                var progress = (float)i / migrationObjects.Count;
+                if (EditorUtility.DisplayCancelableProgressBar("Migration Tool", $"Migrating all {type.Name} components from selection", progress))
+                {
+                    break;
+                }
+                string assetPath = AssetDatabase.GetAssetPath(migrationObjects[i]);
+
+                // Object migrationObject is a scene game object
+                if (String.IsNullOrEmpty(assetPath))
+                {
+                    MigrateGameObjectHierarchy((GameObject)migrationObjects[i]);
+                }
+                else
+                {
+                    if (migrationObjects[i] is GameObject)
+                    {
+                        MigratePrefab(assetPath);
+                    }
+                    else if (migrationObjects[i] is SceneAsset)
+                    {
+                        MigrateScene(assetPath);
+                    }
+                }
+            }
+            EditorUtility.ClearProgressBar();
+            migrationObjects.Clear();
+
+            if (!String.IsNullOrEmpty(previousScenePath) && previousScenePath != EditorSceneManager.GetActiveScene().path)
+            {
+                EditorSceneManager.OpenScene(Directory.GetCurrentDirectory() + "/" + previousScenePath);
```

Comment:
Prefer using Path APIs over manually joining paths yourself, so that you can avoid creating odd/malformed ones (or ones that don't work on various platforms):

https://docs.microsoft.com/en-us/dotnet/api/system.io.path.join?view=netcore-3.1&viewFallbackFrom=netframework-4.8